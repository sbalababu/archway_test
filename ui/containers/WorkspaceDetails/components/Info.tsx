import * as React from 'react';
import { Icon } from 'antd';

interface Props {
  behavior: string;
  name: string;
  summary: string;
}

const Info = ({ behavior, name, summary }: Props) => (
  <div style={{ textAlign: 'center' }}>
    <Icon style={{ fontSize: 72 }} type={behavior === 'simple' ? 'team' : 'deployment-unit'} />
    <div style={{ fontSize: 32 }}>{name}</div>
    <div style={{ fontSize: 16, marginTop: -4 }}>{summary}</div>
  </div>
);

export default Info;
