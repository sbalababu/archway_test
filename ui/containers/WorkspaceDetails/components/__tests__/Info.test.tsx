import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Info from '../Info';

describe('Info', () => {
  it('renders correctly', () => {
    const props = {
      behavior: 'info',
      name: 'rkaland',
      summary: 'info-rkaland',
    };
    const wrapper = shallow(<Info {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
