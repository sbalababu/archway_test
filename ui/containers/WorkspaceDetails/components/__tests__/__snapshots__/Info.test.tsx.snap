// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info renders correctly 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <Icon
    style={
      Object {
        "fontSize": 72,
      }
    }
    type="deployment-unit"
  />
  <div
    style={
      Object {
        "fontSize": 32,
      }
    }
  >
    rkaland
  </div>
  <div
    style={
      Object {
        "fontSize": 16,
        "marginTop": -4,
      }
    }
  >
    info-rkaland
  </div>
</div>
`;
